src/controls/adaptiveCardHost/AdaptiveCardHostHelpers.ts

```diff
@@ -1024,4 +1025,27 @@ const setCSSVariables = (domElement: HTMLElement, prefix: string, obj: any) => {
             domElement.style.setProperty(`--${prefix}-${key}`, obj[key]);
         });
     }
+};
+
+export const injectContextProperty = (dataObject: { "$root": object }, theme: ITheme, spfxContext: BaseComponentContext): IEvaluationContext => {
+    let evaluationContext: IEvaluationContext;
+    let context = {
+        theme: theme,
+        aadInfo: (spfxContext?.pageContext?.aadInfo) ? JSON.parse(JSON.stringify(spfxContext?.pageContext?.aadInfo)) : undefined,
```

Comment:
`JSON.stringify` on classes may lead to issues in future if there some new properties-instances with circular dependencies.
Not ideal way, but not buggy, is to directly specify what properties from the class you want to copy to your context object.

Reply:
Hi @AJIXuMuK , I understand your request of changes and you are right. Let me understand a different way to do that, and I'll push the updates.

Reply:
Hi @AJIXuMuK , I followed your suggestion and fixed how to retrieve and use context properties. This way there are no problems with serialization. Thanks!